core/Array.carp

```diff
@@ -315,9 +315,11 @@ The trailing elements of the longer array will be discarded.")
 
   (doc enumerated "creates a new array of `Pair`s where the first position is the index and the second position is the element from the original array `xs`.")
   (defn enumerated [xs]
-    (zip &Pair.init-from-refs
-         &(range 0 (length xs) 1) ; Inefficient, creates a temporary array.
-         xs))
+    (if (Array.empty? xs) ; Only there because range will crash with a length of 0
```

Comment:
We could also get rid of the intermediate array and this special case if we transform this to an explicit `for` loop. This is not necessarily nicer, but it would solve both problems at once.

Reply:
Yeah a more procedural approach was what I had in mind to remove range and not have it's new signature "contaminate" enumerated. Went with the quickest here. I'll try to update the PR later.

Reply:
Yeah, no worries!

Reply:
Updated.